public/app/core/utils/EventsProvider.tsx

```diff
@@ -0,0 +1,101 @@
+import React, { ComponentType, FunctionComponent, useEffect } from 'react';
+import { Observable, Subject, Unsubscribable } from 'rxjs';
+import { filter } from 'rxjs/operators';
+import { AppEvent, Subtract } from '@grafana/data';
+
+import appEvents from '../app_events';
+
+export interface EventsContextApi<SubscribeEvents, PublishEvents> {
+  events: (originFilter: string) => Observable<AppEvent<SubscribeEvents>>;
+  publish: (event: AppEvent<PublishEvents>, origin: string, payload?: PublishEvents) => void;
+}
+
+export interface RootEventsContextApi {
+  rootEvents: Observable<AppEvent<any>>;
+  rootPublish: (event: AppEvent<any>, payload?: any) => void;
+}
+
+const events = new Subject<AppEvent<any>>();
+const publish = <T extends {} = {}>(event: AppEvent<T>, payload?: T) => {
+  events.next({ ...event, payload });
+  appEvents.emit(event, payload);
+};
+
+export const rootEventing: RootEventsContextApi = {
+  rootEvents: events,
```

Comment:
Why the `root` in the name? Is there any case when it won't be `root`?

Reply:
I could change that